pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -82,10 +82,14 @@ class Policy(base.BasePolicy):
         queue (~queue.Queue): (Optional.) A Queue instance, appropriate
             for crossing the concurrency boundary implemented by
             ``executor``.
+        ack_on_receipt (boolean): 
+            (Optional.) Whether to send `modify_ack_deadlines` upon 
+            receipt first at the cost of increased memory or to send receipts
```

Comment:
_Quo ante_, but _if_ this knob is to exist:

```diff
-Whether to send `modify_ack_deadlines` upon
-receipt first at the cost of increased memory or to send receipts
-as the messages get processed by callbacks.
+Whether to send a `modify_ack_deadline` upon receipt of each message.
+The server uses this to know when to start the initial re-delivery timer.
```

The memory usage is trivial and the server behavioral difference is critical.

Reply:
As long as it doesn't blow up memory I'm fine not having the knob also.